Riot/Modules/Home/HomeViewController.m

```diff
@@ -793,4 +817,72 @@ - (void)setupCrossSigningWithTitle:(NSString*)title
      }];
 }
 
+#pragma mark - Empty view management
+
+- (void)showEmptyView:(BOOL)show
+{
+    if (show && !self.homeEmptyView)
+    {
+        HomeEmptyView *homeEmptyView = [HomeEmptyView instantiate];
+        [homeEmptyView updateWithTheme:ThemeService.shared.theme];
+        [self addEmptyView:homeEmptyView];
+        
+        self.homeEmptyView = homeEmptyView;
+        
+        [self updateEmptyViewDisplayName];
+    }
+    else if (!show)
+    {
+        [self.homeEmptyView removeFromSuperview];
+    }
+    
+    self.recentsTableView.hidden = show;
+}
+
+- (void)updateEmptyViewDisplayName
+{
+    MXUser *myUser = self.mainSession.myUser;
+    NSString *displayName = myUser.displayname ?: myUser.userId;
+    
+    [self.homeEmptyView fillWith:displayName ?: @""];
+}
+
+- (void)addEmptyView:(UIView*)emptyView
+{
+    [self.view insertSubview:emptyView belowSubview:plusButtonImageView];
+
+    emptyView.translatesAutoresizingMaskIntoConstraints = NO;
+    
+    [NSLayoutConstraint activateConstraints:@[
+        [emptyView.topAnchor constraintEqualToAnchor:emptyView.superview.topAnchor],
+        [emptyView.leftAnchor constraintEqualToAnchor:emptyView.superview.leftAnchor],
+        [emptyView.rightAnchor constraintEqualToAnchor:emptyView.superview.rightAnchor],
+        [emptyView.bottomAnchor constraintEqualToAnchor:plusButtonImageView.topAnchor]
+    ]];
+}
+
+// By default on fresh account
+- (BOOL)shouldShowEmptyView
+{
+    // Check if some banners should be displayed
+    if (recentsDataSource.secureBackupBannerSection != -1 || recentsDataSource.crossSigningBannerSection != -1)
+    {
+        return YES;
```

Comment:
Why do we need to check those banners? Why not checking only self. totalItemCounts ?

Plus I am not sure about the YES. I think NO is better.
If you have the secureBackupBannerSection, you have rooms. So, the room list is not empty
If you have crossSigningBannerSection, you are already on another device , you can have (and you probably) rooms.

Reply:
Yes it's an edge case. Maybe you added one encrypted room and then remove it. You can see the empty view and no more theses banners.